internal/status/advl4_status.go

```diff
@@ -160,9 +200,40 @@ func updateGatewayStatusObject(gw *advl4v1alpha1pre1.Gateway, updateStatus *advl
 			utils.AviLog.Warnf("gateway not found %v", err)
 			return err
 		}
-		return updateGatewayStatusObject(updatedGW, updateStatus, retry+1)
+		return UpdateGatewayStatusObject(updatedGW, updateStatus, retry+1)
 	}
 
 	utils.AviLog.Infof("msg: Successfully updated the gateway %s/%s status %+v", gw.Namespace, gw.Name, utils.Stringify(updateStatus))
 	return nil
 }
+
+func InitializeGatewayConditions(gw *advl4v1alpha1pre1.Gateway) error {
```

Comment:
If AKO is rebooted, we would get a state transition from init-->ready--> etc...and the last updated timestamp would be modified. I think that's fine because think in the `pod` case, if the Kubernetes API dies, it would kill the pod and bring it back up when the controller is online?